src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final Random RANDOM = new Random();
+    public static final int FIGURE_COUNT = 5;
+    public static final int MAX_BOUND = 30;
+    public static final int RADIUS = 10;
+
+    private int getBound() {
+        return RANDOM.nextInt(MAX_BOUND) + 1;
+    }
+
+    public Figure getRandomFigure() {
+        int figure = RANDOM.nextInt(FIGURE_COUNT) + 1;
+        switch (figure) {
+            case 1: {
+                return new Circle(new ColorSupplier().getRandomColor(),
```

Comment:
make `ColorSupplier` class level-variable, and reuse it, instead of creating a new one every single time